feat: 장소 검색 백엔드 구현

hyelin0111 opened the pull request:
1. POST/api/searches : api를 통해 장소 검색 및 db에 저장
	- OK : 장소 검색 성공
2. pom.xml json dependency 추가

hyelin0111 committed:
feat: 장소 검색 백엔드 구현

1. POST/api/searches : api를 통해 장소 검색 및 db에 저장
	- OK : 장소 검색 성공
2. pom.xml json dependency 추가

vsh123 commented on src/main/java/com/kvp/thinkjeonju/controller/SearchController.java:
```diff
@@ -0,0 +1,42 @@
+		ArrayList<SpotDTO> spots = searchService.searchSpot(dataValue);
+		/*for(int i=0; i<spots.size(); i++) {
+			System.out.println(spots.get(i).getImgUrl());
+		}*/
```
안쓰는 부분이라면 삭제하는 것도 좋을듯 싶어요!

hyelin0111 replied:
네~

mindock commented on src/main/java/com/kvp/thinkjeonju/controller/HomeController.java:
```diff
@@ -0,0 +1,13 @@
+public class HomeController {
+	
+	@GetMapping("")
+	public String home() { return "spotData";}
```
혹시 이 spotData 안쓰는 거면 index로 고쳐주세요!
/ -> 시작 홈페이지인데 spotData가 나오면 안될거같아서!

hyelin0111 replied:
이거는 내가 혼자 테스트해본다고 만들었던 페이진데 수정을 안했어. 수정할게!

mindock commented on src/main/java/com/kvp/thinkjeonju/controller/SearchController.java:
```diff
@@ -0,0 +1,42 @@
+		log.debug("[Search] Spot Data 검색");
+		
+		// 키워드로 api 통해 spot 검색 후 db에 저장
+		searchService.getSpotData(dataValue);
```
getSpotData보다는 storeSpotData 이런게 더 맞지 않을까?
아니면 api에서 spot 데이터를 가져오는 것 따로 저장하는거 따로 이렇게?

hyelin0111 replied:
수정했어요

mindock commented on src/main/java/com/kvp/thinkjeonju/model/Spot.java:
```diff
@@ -0,0 +1,35 @@
+	}
+	
+	public SpotDTO toDTO() {
+		return new SpotDTO(id, name, content, zipCode, addr, addrDtl, posX, posY, url, tel, fileCnt, null);
```
ImgUrl에 null 말고 arrayList를 생성해서 넣는건 어떨까?

hyelin0111 replied:
이거는 그때 물어보고 했던 거 같은데... 수정할게

mindock replied:
맞아...ㅎ 나중에 넣을때를 생각하니깐 저게 좋은거같아서!

mindock commented on src/main/java/com/kvp/thinkjeonju/service/SearchService.java:
```diff
@@ -0,0 +1,207 @@
+public class SearchService {
+
+	final int MAX = 999999;
+	final int IMGMAX = 5;
```
각 객체마다 필요한 값이 아닌 경우, 앞에 static을 붙여야하지 않을까?

hyelin0111 replied:
네~

mindock commented on src/main/java/com/kvp/thinkjeonju/service/SearchService.java:
```diff
@@ -0,0 +1,207 @@
+	final int MAX = 999999;
+	final int IMGMAX = 5;
+	
+	private String authApiKey = "DCSBMCAPSGRWVRX";
```
이 값도 바뀌지 않는다면, final을 붙여야하지 않을까??
안 붙인 이유는?

hyelin0111 replied:
수정했어!

mindock commented on src/main/java/com/kvp/thinkjeonju/service/SearchService.java:
```diff
@@ -0,0 +1,207 @@
+    					obj.getString("addrDtl"), obj.getDouble("posx"), obj.getDouble("posy"), 
+    					obj.getString("userHomepage"), obj.getString("tel"), obj.getInt("fileCnt"), null));
+    			} else { // 길이가 2이상일 때
+    				JSONArray spot = (JSONArray)obj.get("list");
```
totalCount <= 1인 경우의
obj = (JSONObject)obj.get("list");
여기랑 같은 거니깐 if문 위로 빼는 건 어떨까?

hyelin0111 replied:
totalCount <= 1인 경우는 list를 JSONObject로 변환한거고,
else문에서는 list를 JSONArray로 변환한거야.

mindock commented on src/main/java/com/kvp/thinkjeonju/service/SearchService.java:
```diff
@@ -0,0 +1,207 @@
+			if(spot.get(i).getFileCnt() != 0) {
+				ArrayList<String> img = searchMapper.searchSpotImg(spot.get(i).getId());
+				spot.get(i).setImgUrl(img);
+				//System.out.println(spot.get(i).getImgUrl());
```
주석지우기!

hyelin0111 replied:
네~

mindock commented on src/main/java/com/kvp/thinkjeonju/service/SearchService.java:
```diff
@@ -0,0 +1,207 @@
+	public void addSpotImg(SpotImgDTO spotImgDTO) {
+		int result = searchMapper.addSpotImg(spotImgDTO);
+		/*if(result <= 0) 
+			throw new DatabaseException("장소 이미지 등록에 실패했습니다.");*/
```
pull 받아서 예외 처리하기

hyelin0111 replied:
넵

mindock commented on src/main/java/com/kvp/thinkjeonju/service/SearchService.java:
```diff
@@ -0,0 +1,207 @@
+        			if(checkSpotIdDuplicate(spots.get(i).getId()) == 0) {	
+        				addSpot(spots.get(i));
+        				if(spots.get(i).getFileCnt() != 0) {				  
+            				getSpotImgUrl(spots.get(i).getId());
```
들여쓰기?

hyelin0111 replied:
이거 내꺼에는 제대로 되어있는데?

mindock reviewed (commented):
현재 
- API 검색 -> 리스트 얻음 -> DB에 존재하지 않는 SPOT 저장 -> SPOT_IMG 저장
 -> DB에서 검색 -> 리스트 반환

내 생각
- API 검색 -> 리스트 얻음 -> DB에 존재하지 않는 SPOT 저장 -> SPOT_IMG 저장 -> API 얻은 리스트 반환
이와 같은 형태로 바꾸는 것에 대해 어떻게 생각하는지??

DB에 저장한 SPOT정보는 좋아요 또는 장소 메뉴에서 보여줄 때 사용하는 것만 사용해도 되지않을까?
라는 생각을 했는데 둘의 의견도 듣고 싶음!

hyelin0111 committed:
refactor: 장소 검색 api 코드 수정

1. api결과를 바로 리턴하게 수정
	API 검색 -> 리스트 얻음 -> DB에 존재하지 않는 SPOT 저장 -> SPOT_IMG 저장 -> API 얻은 리스트 반환

2. SearchController -> SpotController 등 파일명 변경
3. 불필요한 코드 삭제

hyelin0111 committed:
Merge branch 'develop' of https://github.com/kvp9356/think-jeonju into
feat/search

# Conflicts:
#	bin/src/main/resources/mybatis-config.xml
#	src/main/java/com/kvp/thinkjeonju/controller/HomeController.java
#	src/main/java/com/kvp/thinkjeonju/exception/common/DataBaseException.java
#	src/main/resources/templates/index.html

hyelin0111 committed:
refactor: bin 폴더 삭제
